File: classroomApplication/src/main/java/hu/elte/alkfejl/classroomApplication/model/Room.java
package hu.elte.alkfejl.classroomApplication.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "ROOMS")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Room extends BaseEntity{

    @Column(nullable = false, unique = true)
    private String name;

    public enum RoomStatus {
        RESERVED, FREE
    }
}

File: classroomApplication/classroomApplication/src/app/model/Room.ts
export class RoomStatus {
  static RESERVED: String = "RESERVED";
  static FREE: String = "FREE";
}

export class Room {

    id: number;
    name: String;
    status: RoomStatus;

    constructor(name?: String, status?: RoomStatus, id?: number){
        this.name = name;
        this.id = id;
        this.status = status || '';
    }
}
